hackathon_site/dashboard/frontend/src/components/inventory/ProductOverview/ProductOverview.test.js

```diff
@@ -65,15 +69,37 @@ describe("<EnhancedAddToCartForm />", () => {
 
     test("requestFailure is passed in and rendered properly", () => {
         const requestFailure = { message: "Failed" };
-        const availableQuantity = "4";
+        const quantityAvailable = 4;
 
         const { getByText } = render(
             <EnhancedAddToCartForm
                 requestFailure={requestFailure}
-                availableQuantity={availableQuantity}
+                quantityAvailable={quantityAvailable}
             />
         );
 
         expect(getByText(requestFailure.message)).toBeInTheDocument();
     });
+
+    test("button and select are disabled if quantityAvailable is 0", () => {
+        const { getByText, getByTestId } = render(
+            <EnhancedAddToCartForm quantityAvailable={0} />
+        );
+
+        const button = getByText("Add to cart");
+        const select = getByTestId("qtySelect");
+
+        expect(button).toBeDisabled();
+        expect(select).toBeDisabled();
+    });
+
+    test("dropdown values are minimum between quantityAvailable and maxConstraint", () => {
+        const { queryByText, getByRole } = render(
+            <EnhancedAddToCartForm quantityAvailable={3} constraintMax={2} />
+        );
+
+        fireEvent.mouseDown(getByRole("button", { name: "1" }));
+
+        expect(queryByText("3")).not.toBeInTheDocument();
```

Comment:
This would still pass if the component returned nothing at all. Adding an extra check that "2" is in the document would fix that (I tested locally and that does pass).